### ISSUE TITLE
set --outFilterScoreMinOverLread and outFilterMatchNminOverLread to be 0

### ISSUE THREAD

**Issue by colinwxl** [alexdobin/STAR#415]
I used STAR aligner to map human RNA-Seq fastq data to reference genome. As I had trimmed the fastq data previously, so I encounter huge percentage of "% of reads unmapped: too short". I learned from https://github.com/alexdobin/STAR/issues/169. Setting set `--outFilterScoreMinOverLread` and `--outFilterMatchNminOverLread` to be 0, while setting `--outFilterMismatchNmax` 2 did make "% of reads unmapped: too short" to be small. However, the " % of reads mapped to multiple loci" increased.
I wondered it is reasonable to set `--outFilterScoreMinOverLread 0` and `--outFilterMatchNminOverLread 0`, while set `--outFilterMismatchNmax` 2. Hope you can deliver some explanation. Thanks!
@alexdobin @bostanict

My raw fastq data is 150nt per read pair-end's data. I had kept those reads whose length was greater than 50nt after quality trimming.

**Comment by elzerac** [comment]
Hello, I am also interested in this answer. I ran into an issue as described in #169 and the fastq order did not seem to be the problem (I re-ordered both the input fastq and also checked the order of the unmapped files). 
However, when I set --outFilterScoreMinOverLread 0 --outFilterMatchNminOverLread 0 --outFilterMatchNmin 0 my unique mapping rate went from ~30% to ~80% ! And % of reads unmapped: too short went down to 0%. 

However, I am now wondering whether I should also include the flag --outFilterMismatchNmax 2 as described in #169 . Any advice would be very much appreciated! 

Thanks, 

Elze

**Comment by gaoyy** [comment]
hi, I have the data set from human, however , it has contaminated with  Bacillus subtilis strain ,when i set `-outFilterScoreMinOverLread 0 --outFilterMatchNminOverLread 0 `, the mapping rate  is nearly 98% ,then I set  ` -outFilterScoreMinOverLread 0.3 --outFilterMatchNminOverLread 0.3` ,the mapping rate become 0.52%

**Comment by alexdobin** [comment]
Hi All,

sorry for the belayed reply. Do not hesitate to bump up your posts if I do not answer within a week.

Setting --outFilterScoreMinOverLread 0 --outFilterMatchNminOverLread 0 removes any limits on the mapped length, so even if short portions of the reads align, the reads will be considered mapped. This explains why you are seeing high mapping rate - albeit such short alignments are often multimappers. I do not think this is generally a good idea, as such short alignments will have a high rate of wrong alignments. It is better to figure out why the mappability is poor, e.g.
1. File formatting issues
2. Poor sequencing quality
3. Contamination
4. Sample divergence from the reference genome

Cheers
Alex

**Comment by colinwxl** [comment]
@elzerac Could you share your quality trimming strategies? Did you mean that you limited longer "passed" reads' length?

**Comment by elzerac** [comment]
@colinwxl I was trying to map an extremely poor quality run to see if I could salvage any data from the run itself. I ended up trimming 150bp reads to roughly 50bp using Trimgalore and before running STAR SE alignment (discarding Read2 completely due to poor quality). It was a less than ideal situation, that I would not recommend doing, but was trying to get something out of that sequencing run. Re-making my libraries improved my run quality and mapping rate tremendously (I ended up using the default ENCODE recommended flags, with resulting >90% unique mapping rate).
